drivers/can/CMakeLists.txt

```diff
@@ -12,3 +12,15 @@ zephyr_sources_ifdef(CONFIG_CAN_RCAR         can_rcar.c)
 zephyr_sources_ifdef(CONFIG_USERSPACE        can_handlers.c)
 zephyr_sources_ifdef(CONFIG_CAN_SHELL        can_shell.c)
 zephyr_sources_ifdef(CONFIG_CAN_NET          can_net.c)
+
+if(CONFIG_CAN_NATIVE_POSIX)
```

Comment:
Just a quick comment: socket CAN is only available on Linux, so I think a name other than "native_posix" should be used. The build should only be available on Linux hosts actually (maybe `CMAKE_HOST_SYSTEM_NAME` can be useful here?)